backend/src/realtime/realtime-note/realtime-user-status-adapter.spec.ts

```diff
@@ -17,70 +17,72 @@ type SendMessageSpy = jest.SpyInstance<
 >;
 
 describe('realtime user status adapter', () => {
-  let client1: RealtimeConnection;
-  let client2: RealtimeConnection;
-  let client3: RealtimeConnection;
-  let client4: RealtimeConnection;
+  let clientSource: RealtimeConnection;
+  let clientLoggedIn: RealtimeConnection;
+  let clientGuest: RealtimeConnection;
+  let clientNotReady: RealtimeConnection;
 
-  let sendMessage1Spy: SendMessageSpy;
-  let sendMessage2Spy: SendMessageSpy;
-  let sendMessage3Spy: SendMessageSpy;
-  let sendMessage4Spy: SendMessageSpy;
+  let sendMessageSource: SendMessageSpy;
+  let sendMessageLoggedIn: SendMessageSpy;
+  let sendMessageGuest: SendMessageSpy;
+  let sendMessageNotReady: SendMessageSpy;
 
   let realtimeNote: RealtimeNote;
 
-  const username1 = 'mock1';
-  const username2 = 'mock2';
-  const username3 = 'mock3';
-  const username4 = 'mock4';
+  const usernameSource = 'source.sender';
+  const usernameLoggedIn = 'logged.in';
+  const usernameNotReady = 'not.ready';
+
+  function spyOnSendMessage(connection: RealtimeConnection): jest.SpyInstance {
+    return jest.spyOn(connection.getTransporter(), 'sendMessage');
+  }
 
   beforeEach(() => {
     realtimeNote = new RealtimeNote(
       Mock.of<Note>({ id: 9876 }),
       'mockedContent',
     );
-    client1 = new MockConnectionBuilder(realtimeNote)
-      .withRealtimeUserState()
-      .withUsername(username1)
+    clientSource = new MockConnectionBuilder(realtimeNote)
+      .withRealtimeUserStatus()
+      .withLoggedInUser(usernameSource)
       .build();
-    client2 = new MockConnectionBuilder(realtimeNote)
-      .withRealtimeUserState()
-      .withUsername(username2)
+    clientLoggedIn = new MockConnectionBuilder(realtimeNote)
+      .withRealtimeUserStatus()
+      .withLoggedInUser(usernameLoggedIn)
       .build();
-    client3 = new MockConnectionBuilder(realtimeNote)
-      .withRealtimeUserState()
-      .withUsername(username3)
+    clientGuest = new MockConnectionBuilder(realtimeNote)
+      .withRealtimeUserStatus()
+      .withGuestUser('Virtuous Mockingbird')
       .build();
-    client4 = new MockConnectionBuilder(realtimeNote)
-      .withRealtimeUserState()
-      .withUsername(username4)
+    clientNotReady = new MockConnectionBuilder(realtimeNote)
+      .withRealtimeUserStatus()
+      .withLoggedInUser(usernameNotReady)
       .build();
 
-    sendMessage1Spy = jest.spyOn(client1.getTransporter(), 'sendMessage');
-    sendMessage2Spy = jest.spyOn(client2.getTransporter(), 'sendMessage');
-    sendMessage3Spy = jest.spyOn(client3.getTransporter(), 'sendMessage');
-    sendMessage4Spy = jest.spyOn(client4.getTransporter(), 'sendMessage');
+    sendMessageSource = spyOnSendMessage(clientSource);
```

Comment:
I think these names are a bit confusing. Maybe we should either use `sendMessageSourceClient` or `clientSourceSendMessage`?